docs/content/guides/2.alokai-essentials/1.alokai-next-js/9.udl.md

```diff
@@ -471,20 +472,88 @@ export default function ProductDetails({ product }: ProductDetailsProps) {
 Finally, if you run the application, you will see that both home page and product details page are working as expected. Except for one thing. If you try to add a product to the cart, you will see that we have an issue with the `useCart` hook - nothing happens when you click the "Add to cart" button.
 
-I will leave this as an exercise for you to fix. You can use the approach we learned above to fix the `useCart` hook. Only, I will give you a small hint - you need to replace the `addToCart` method from the `sapcc` module with the `addToCart` method from the `unified` module and as well, you need to replace the `Cart` type with the `SfCart` type from the Unified Data Model.
+I will leave this as an exercise for you to fix. You can use the approach we learned above to fix the `useCart` hook. Only, I will give you a small hint - you need to replace the `addToCart` method from the `sapcc` module with the `addToCart` method from the `unified` module and as well, you need to replace the `Cart` type with the `SfCart` type from the Unified Data Model. You can also find the solution here [udl branch](https://github.com/vuestorefront-community/nextjs-starter/tree/udl).
 
 You can find more information about different Unified Api Methods in the [Unified Cart Methods](https://docs.vuestorefront.io/storefront/unified-data-layer/unified-methods/cart) section of Storefront documentation.
 
+## Displaying Images
+
+So far we've been displaying hardcoded images. Now thanks to the Unified Data Layer we can easily access images from the e-commerce platform.
+
+Open the `storefront/components/ProductGallery.tsx` file and add the following code:
+
+```tsx
+// ... rest of the code
+import { Product } from '@vsf-enterprise/sap-commerce-webservices-sdk';
+
+interface ProductGalleryProps {
+  images: SfProduct['gallery'];
+}
+// ... rest of the code
+```
+
+In the above code, we have created a TypeScript interface `ProductGalleryProps` for the props of the `ProductGallery` Block. We have used the `SfProduct` type from the UDL.
+
+Now, we will use the `ProductGalleryProps` interface to define the type of the `props` of the `ProductGallery` component. Replace the content of the `storefront/components/ProductGallery.tsx` file with the following code:
+
+```diff
+- export default function GalleryVertical() {
++ export default function GalleryVertical({ images }: ProductGalleryProps) {
+```
+
+We can also remove the `images` constant and `withBase` function from the `ProductGallery` component as we are now passing the `images` prop from the parent component.
+
+Now, Replace all the occurrences of `imageThumbSrc` and `imageSrc` with `url`. The final shape of the `ProductGallery` component can be found in the [udl branch](https://github.com/vuestorefront-community/nextjs-starter/tree/udl).
+
+:::info
+Your IDE might yell that we're using `<img>` instead of `next/image`. That's because we've copied the Gallery component from Storefront UI  which is agnostic of a meta framework.
```

Comment:
```suggestion
Your IDE might throw a warning that we're using `<img>` instead of `next/image`. That's because we've copied the Gallery component from Storefront UI  which is agnostic of a meta framework.
```